### PR TITLE
Fix some small type issues.

### LINKED ISSUE #156: Small type issues
Hi.  Thanks for adding types to hyperlink!  I noticed 2 smallish improvements that could be made.

1. It would be nice if `parse_url` used `typing.overload` to determine which type of URL it returns (i.e. DecodedURL vs URL).
2. I believe QueryParameters could just be typed as:
```
QueryParameters = Iterable[Tuple[Text, Optional[Text]]
```
(This way it can also accepts generators and comprehensions)

Let me know if you'd like me to make a PR with these changes.

### PR DESCRIPTION
**Pull request by euresti**
Use @overload to have a better return type for parse.
Use Iterable for QueryParameters to support generators and views.

Fixes #156

### CODE REVIEW

**Comment by euresti** [comment]
I think the appveyor errors are caused by pip no longer supporting 2.7 nor 3.5.

File: src/hyperlink/_url.py [hunk 1]
```diff
@@ -2415,6 +2420,25 @@ def __dir__(self):
     # # End Twisted Compat Code
 
 
+# Add some overloads so that parse gives a better return value.
+@overload
+def parse(url, decoded, lazy=False):
+    # type: (Text, Literal[False], bool) -> URL
```
Comment (wsanchez) [inline src/hyperlink/_url.py:2426]: neat-o

**Review by wsanchez — APPROVED** [review]
Awesome, thanks

**Comment by wsanchez** [comment]
@mahmoud @glyph Any objection to dropping 2.7 and 3.5 on Windows from the build matrix?

**Comment by glyph** [comment]
None from me! Into the memory hole.

**Pull request merged**